From 96d0580ec0e0d0586272553cc56fda082cf96925 Mon Sep 17 00:00:00 2001
From: Phil Smith <phil.h.smith@gmail.com>
Date: Sat, 20 Mar 2010 12:22:24 -0700
Subject: [PATCH] Removing namespace components from NamedBase#table_name. [#4230 state:resolved]

table_name now omits any leading namespace parts.  Generators using
tablename haven't been changed.  Particularly, this implies that
migrations created by the model generator no longer include the
namespace components in their filenames.
---
 railties/lib/rails/generators/named_base.rb      |    5 +----
 railties/test/generators/model_generator_test.rb |   18 +++++++++---------
 railties/test/generators/named_base_test.rb      |    6 +++---
 3 files changed, 13 insertions(+), 16 deletions(-)

diff --git a/railties/lib/rails/generators/named_base.rb b/railties/lib/rails/generators/named_base.rb
index 8d1dfbd..1420589 100644
--- a/railties/lib/rails/generators/named_base.rb
+++ b/railties/lib/rails/generators/named_base.rb
@@ -40,10 +40,7 @@ module Rails
         end
 
         def table_name
-          @table_name ||= begin
-            base = pluralize_table_names? ? plural_name : singular_name
-            (class_path + [base]).join('_')
-          end
+          @table_name ||= pluralize_table_names? ? plural_name : singular_name
         end
 
         # Tries to retrieve the application name or simple return application.
diff --git a/railties/test/generators/model_generator_test.rb b/railties/test/generators/model_generator_test.rb
index 79ce9a2..124ed96 100644
--- a/railties/test/generators/model_generator_test.rb
+++ b/railties/test/generators/model_generator_test.rb
@@ -34,16 +34,16 @@ class ModelGeneratorTest < Rails::Generators::TestCase
 
   def test_migration_with_namespace
     run_generator ["Gallery::Image"]
-    assert_migration "db/migrate/create_gallery_images", /class CreateGalleryImages < ActiveRecord::Migration/
-    assert_no_migration "db/migrate/create_images"
+    assert_no_migration "db/migrate/create_gallery_images"
+    assert_migration "db/migrate/create_images", /class CreateImages < ActiveRecord::Migration/
   end
 
   def test_migration_with_nested_namespace
     run_generator ["Admin::Gallery::Image"]
-    assert_no_migration "db/migrate/create_images"
+    assert_no_migration "db/migrate/create_admin_gallery_images"
     assert_no_migration "db/migrate/create_gallery_images"
-    assert_migration "db/migrate/create_admin_gallery_images", /class CreateAdminGalleryImages < ActiveRecord::Migration/
-    assert_migration "db/migrate/create_admin_gallery_images", /create_table :admin_gallery_images/
+    assert_migration "db/migrate/create_images", /class CreateImages < ActiveRecord::Migration/
+    assert_migration "db/migrate/create_images", /create_table :images/
   end
 
   def test_migration_with_nested_namespace_without_pluralization
@@ -52,8 +52,8 @@ class ModelGeneratorTest < Rails::Generators::TestCase
     assert_no_migration "db/migrate/create_images"
     assert_no_migration "db/migrate/create_gallery_images"
     assert_no_migration "db/migrate/create_admin_gallery_images"
-    assert_migration "db/migrate/create_admin_gallery_image", /class CreateAdminGalleryImage < ActiveRecord::Migration/
-    assert_migration "db/migrate/create_admin_gallery_image", /create_table :admin_gallery_image/
+    assert_migration "db/migrate/create_image", /class CreateImage < ActiveRecord::Migration/
+    assert_migration "db/migrate/create_image", /create_table :image/
   ensure
     ActiveRecord::Base.pluralize_table_names = true
   end
@@ -61,8 +61,8 @@ class ModelGeneratorTest < Rails::Generators::TestCase
   def test_migration_with_namespaces_in_model_name_without_plurization
     ActiveRecord::Base.pluralize_table_names = false
     run_generator ["Gallery::Image"]
-    assert_migration "db/migrate/create_gallery_image", /class CreateGalleryImage < ActiveRecord::Migration/
-    assert_no_migration "db/migrate/create_gallery_images"
+    assert_no_migration "db/migrate/create_images"
+    assert_migration "db/migrate/create_image", /class CreateImage < ActiveRecord::Migration/
   ensure
     ActiveRecord::Base.pluralize_table_names = true
   end
diff --git a/railties/test/generators/named_base_test.rb b/railties/test/generators/named_base_test.rb
index d30880d..06f68e9 100644
--- a/railties/test/generators/named_base_test.rb
+++ b/railties/test/generators/named_base_test.rb
@@ -40,7 +40,7 @@ class NamedBaseTest < Rails::Generators::TestCase
     assert_name g, 'foo',        :singular_name
     assert_name g, 'foos',       :plural_name
     assert_name g, 'admin.foo',  :i18n_scope
-    assert_name g, 'admin_foos', :table_name
+    assert_name g, 'foos',       :table_name
   end
 
   def test_named_generator_attributes_as_ruby
@@ -54,13 +54,13 @@ class NamedBaseTest < Rails::Generators::TestCase
     assert_name g, 'Foo',        :human_name
     assert_name g, 'foos',       :plural_name
     assert_name g, 'admin.foo',  :i18n_scope
-    assert_name g, 'admin_foos', :table_name
+    assert_name g, 'foos',       :table_name
   end
 
   def test_named_generator_attributes_without_pluralized
     ActiveRecord::Base.pluralize_table_names = false
     g = generator ['admin/foo']
-    assert_name g, 'admin_foo', :table_name
+    assert_name g, 'foo',       :table_name
   ensure
     ActiveRecord::Base.pluralize_table_names = true
   end
-- 
1.6.6

